core/state/state_object.go

```diff
@@ -242,6 +252,7 @@ func (s *stateObject) SetState(key, value common.Hash) {
 		prevalue: prev,
 	})
 	s.setState(key, value)
+	s.isFilled[key] = false
```

Comment:
Some questions:
- Shouldn't this also happen before L246?
- This strategy seems to assume any witness record is done before a `SetState` call. If this operations are swapped, then we'd be doing the wrong thing? If that's the case sounds like an implicit/dangerous assumption.